frontend/public/components/RBAC/bindings.jsx

```diff
@@ -280,63 +286,95 @@ export const bindingType = (binding) => {
   return binding.metadata.namespace ? 'namespace' : 'cluster';
 };
 
-const roleResources = [
-  { kind: 'RoleBinding', namespaced: true },
-  { kind: 'ClusterRoleBinding', namespaced: false, optional: true },
-];
-
 export const RoleBindingsPage = ({
   namespace = undefined,
   showTitle = true,
   mock = false,
-  staticFilters = undefined,
+  staticFilters,
   name,
   kind,
   createPath = `/k8s/cluster/rolebindings/~new${
     name && kind ? `?subjectName=${name}&subjectKind=${kind}` : ''
   }`,
 }) => {
   const { t } = useTranslation();
-  const rowFilters = [
-    {
-      filterGroupName: t('public~Kind'),
-      type: 'role-binding-kind',
-      reducer: bindingType,
-      itemsGenerator: ({ ClusterRoleBinding: data }) => {
-        const items = [
-          { id: 'namespace', title: t('public~Namespace RoleBindings') },
-          { id: 'system', title: t('public~System RoleBindings') },
-        ];
-        if (data && data.loaded && !data.loadError) {
-          items.unshift({
-            id: 'cluster',
-            title: t('public~Cluster-wide RoleBindings'),
-          });
-        }
-        return items;
-      },
+  const resources = useK8sWatchResources({
+    RoleBinding: {
+      kind: 'RoleBinding',
+      namespaced: true,
+      namespace,
+      isList: true,
     },
-  ];
+    ClusterRoleBinding: {
+      kind: 'ClusterRoleBinding',
+      namespaced: false,
+      isList: true,
+    },
+  });
+
+  const data = React.useMemo(() => flatten(resources), [resources]);
+
+  const loaded = Object.values(resources).every((r) => r.loaded);
+
+  const hasCRBindings =
+    resources.ClusterRoleBinding.data?.length > 0 &&
+    resources.ClusterRoleBinding.loaded &&
+    !resources.ClusterRoleBinding.loadError;
+
+  const rowFilters = React.useMemo(
+    () => [
+      {
+        filterGroupName: t('public~Kind'),
+        type: 'role-binding-kind',
+        reducer: bindingType,
+        filter: (filter, binding) =>
+          filter.selected?.includes(bindingType(binding)) || filter.selected?.size === 0,
```

Comment:
nit: The `filter.selected?.size === 0` could be written as `!filter.selected?.length`.

Reply:
updated